app/templates/_gulpfile.js

```diff
@@ -559,8 +559,9 @@ gulp.task('deploy:demo', (cb) => {
 /////////////////////////////////////////////////////////////////////////////
 gulp.task('test', (cb) => {<% if(testingFramework === 'jest') {%>
   let isTravis = !!process.env.TRAVIS;
-  jestCli.runCLI({ config: require('./package.json').jest, coverage: true, runInBand: isTravis, ci: isTravis }, ".", result => 
-    cb(result.success ? undefined: 'There are test failures!'));<% } else { %>
+  return jestCli.runCLI({ config: require('./package.json').jest, coverage: true, runInBand: isTravis, ci: isTravis }, ".", result => {
```

Comment:
> Also task sequence seem to be stopped once reach any test task (test, test:xxx)
> You can reproduce this by:
> * Generate Ng2 project using Jest test framework
> * Run gulp build:watch inside generated project

Good catch man! 👍 ).

But your solution is still not correct.
In fact, with it, a failing test *WILL NOT* stop the sequence, because the `Promise` returned by `runCLI()`will always resolved (so gulp will continue with next task, even if there are test failures).

Instead, the `cb()`callback just has to be called properly (i.e when `'runCLI()` promise resolves):

```
gulp.task('test', (cb) => {
  let isTravis = !!process.env.TRAVIS;
  jestCli.runCLI({ config: require('./package.json').jest, coverage: true, runInBand: isTravis, ci: isTravis }, ".").then(({ results }) => {
    cb(results.success ? undefined: 'There are test failures!')
  });
});
```

Same goes for the other `'test:xxx'` tasks